### PR TITLE
Include delete query TCK [3/5]

### PR DESCRIPTION
**Pull request by otaviojava**
This pull request adds comprehensive support for testing delete operations in the `FruitRepository` and its associated integration tests. It introduces new repository methods for various delete queries and a suite of nested test cases to verify their correct behavior, including handling of edge cases and database-specific limitations.

**Repository Enhancements:**

* Added multiple delete query methods to `FruitRepository` for deleting by name, by quantity (with different comparison operators), using `IN`, and with logical `AND`/`OR` conditions.

**Integration Test Improvements:**

* Introduced a new `WhenDeleteQuery` nested test class in `JakartaQueryTests` containing test cases for each new delete method, ensuring coverage for all supported delete operations.
* Each test case includes assertions to verify correct deletion and handles `UnsupportedOperationException` for databases that may not support certain operations.
* Added setup and cleanup logic to maintain test isolation and ensure database consistency after delete operations.

**Test Structure and Readability:**

* Refactored `JakartaQueryTests` to use JUnit 5 features such as `@Nested` and `@DisplayName` for better organization and clearer test output. [[1]](diffhunk://#diff-6543dd8da246f8ec31bb4b5054b9ddbecf75c405f4aef41a763253982791a327R42) [[2]](diffhunk://#diff-6543dd8da246f8ec31bb4b5054b9ddbecf75c405f4aef41a763253982791a327R67-R69)
* Imported additional utilities and annotations to support the new test structure and consistency checks.

### CODE REVIEW

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 1]
```diff
@@ -60,6 +64,9 @@ public void setup() {
         populator.populate(fruitRepository);
     }
 
+    @Nested
+    @DisplayName("When execute the select statement")
+    class WhendSelectQuery {
     @DisplayName("should find all entities as stream")
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]: `WhendSelectQuery` was probably intended to be `WhenSelectQuery`.
The indentation is incorrect here and continues for around 500 lines.
I think it's best not to combine SELECT Query tests and DELETE Query tests into a single test class.  Instead, create a new class called something like JakartaDeleteQueryTests for the new test cases that need to be cleaned up differently and handled differently for eventual consistency.

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
 * Copyright (c) 2026 Contributors to the Eclipse Foundation
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+    @AfterEach
+    public void cleanup() {
+        fruitRepository.deleteAll();
+        TestPropertyUtility.waitForEventualConsistency();
+    }
+
+    @DisplayName("should delete all entities")
+    @Assertion(id = "1318",
+            strategy = "Execute the delete all queries, wait for eventual consistency and verify if all entities are deleted")
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
            strategy = "Execute a deleteAll query, wait for eventual consistency and verify if all entities are deleted")
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    void shouldDeleteAllEntities() {
+
+        try {
+            fruitRepository.deleteAll();
+            TestPropertyUtility.waitForEventualConsistency();
+
+            Assertions.assertThat(fruitRepository.findAll().toList()).isEmpty();
+        } catch (UnsupportedOperationException exp) {
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable of delete all the entities
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]: More precisely, the specification defined limitation is that
```suggestion
                // Column and Key-Value databases might not be capable of querying without a WHERE clause
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete entities using equals condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete not equals condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete greater than condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete greater than equals condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete lesser condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete lesser than equals condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should delete using IN condition")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should test AND")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByNameAndQuantity(fruit.getName(), fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> !(f.getName().equals(fruit.getName())
+                            && f.getQuantity().equals(fruit.getQuantity())));
+        } catch (UnsupportedOperationException exp) {
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
                // Column and Key-Value databases might not be capable of AND.
                // Column and Key-Value databases might not be capable of querying by an attribute that is not the Id.
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should test OR")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+            fruitRepository.deleteByNameOrQuantity(fruit.getName(), fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+            List<Fruit> result = fruitRepository.findAll().toList();
+
+            Assertions.assertThat(result)
+                    .allMatch(f -> !(f.getName().equals(fruit.getName())
+                            || f.getQuantity().equals(fruit.getQuantity())));
+        } catch (UnsupportedOperationException exp) {
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable deleting by attribute that is not a key.
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
                // Column and Key-Value databases might not be capable of OR.
                // Column and Key-Value databases might not be capable querying by an attribute that is not the Id.
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by name equals, verify if entity is deleted")
+    void shouldDeleteEq() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByName(fruit.getName());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> !fruit.getName().equals(f.getName()));
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]: The way this was written, an implementation that has wrong behavior for deletion by name but doesn't support queries without a WHERE clause (failing `.findAll()`) ends up passing the test!  We need to be careful about mixing operations like this into a single test.  A better way to confirm the results would be a query with the same WHERE clause as the DELETE operation to check if the entities are gone. Something like this:
```suggestion
            List<Fruit> result = fruitRepository.findNameEquals(fruit.getName());
            Assertions.assertThat(result)
                    .isEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by name not equals, verify if entity is deleted")
+    void shouldDeleteNeq() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByNotEqualsName(fruit.getName());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> fruit.getName().equals(f.getName()));
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findNameNotEquals(fruit.getName());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findNameEquals(fruit.getName());
            Assertions.assertThat(result2)
                    .isNotEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by quantity greater than, verify if entity is deleted")
+    void shouldGt() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteQuantityGreaterThan(fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> f.getQuantity() <= fruit.getQuantity());
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findQuantityGt(fruit.getQuantity());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findQuantityLte(fruit.getQuantity());
            Assertions.assertThat(result2)
                    .isNotEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 19]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by quantity greater than equals, verify if entity is deleted")
+    void shouldGte() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteQuantityGreaterThanEquals(fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> f.getQuantity() < fruit.getQuantity());
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findQuantityGte(fruit.getQuantity());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findQuantityLt(fruit.getQuantity());
            Assertions.assertThat(result2)
                    .isNotEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by quantity lesser, verify if entity is deleted")
+    void shouldLt() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteLesserThan(fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> f.getQuantity() >= fruit.getQuantity());
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findQuantityLt(fruit.getQuantity());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findQuantityGte(fruit.getQuantity());
            Assertions.assertThat(result2)
                    .isNotEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 21]
```diff
@@ -0,0 +1,282 @@
...
+            strategy = "delete by quantity lesser than equals, verify if entity is deleted")
+    void shouldLte() {
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteQuantityLesserThanEquals(fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> f.getQuantity() > fruit.getQuantity());
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findQuantityLte(fruit.getQuantity());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findQuantityGt(fruit.getQuantity());
            Assertions.assertThat(result2)
                    .isNotEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 22]
```diff
@@ -0,0 +1,282 @@
...
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByNameIn(List.of(fruit.getName(), fruits.get(1).getName()));
+            TestPropertyUtility.waitForEventualConsistency();
+
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> !f.getName().equals(fruit.getName())
+                            ||
+                            !f.getName().equals(fruits.get(1).getName()));
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result1 = fruitRepository.findNameEquals(fruit.getName());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findNameEquals(fruits.get(1).getName());
            Assertions.assertThat(result2)
                    .isEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 23]
```diff
@@ -0,0 +1,282 @@
...
+
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByNameOrQuantity(fruit.getName(), fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+            List<Fruit> result = fruitRepository.findAll().toList();
+
+            Assertions.assertThat(result)
+                    .allMatch(f -> !(f.getName().equals(fruit.getName())
+                            || f.getQuantity().equals(fruit.getQuantity())));
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]: Eliminating `findAll()` here will requiring adding `findQuantityEquals(quantity)` to check results
```suggestion

            List<Fruit> result1 = fruitRepository.findNameEquals(fruit.getName());
            Assertions.assertThat(result1)
                    .isEmpty();
            List<Fruit> result2 = fruitRepository.findQuantityEquals(fruit.getQuantity());
            Assertions.assertThat(result2)
                    .isEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+    void shouldDeleteUsingAndCondition() {
+
+        try {
+            Fruit fruit = fruits.getFirst();
+            fruitRepository.deleteByNameAndQuantity(fruit.getName(), fruit.getQuantity());
+            TestPropertyUtility.waitForEventualConsistency();
+            List<Fruit> result = fruitRepository.findAll().toList();
+            Assertions.assertThat(result)
+                    .allMatch(f -> !(f.getName().equals(fruit.getName())
+                            && f.getQuantity().equals(fruit.getQuantity())));
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java]:
```suggestion
            List<Fruit> result = fruitRepository.findNameEqualsAndQuantityEquals(
                    fruit.getName(),
                    fruit.getQuantity());
            Assertions.assertThat(result)
                    .isEmpty();
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java [hunk 25]
```diff
@@ -0,0 +1,282 @@
...
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(JakartaQueryTests.class,
+                        Fruit.class, FruitRepository.class);
+    }
+
+    @Inject
+    protected FruitRepository fruitRepository;
```
Comment (KyleAure) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java:54]: Fix: use a different entity/repository than `Fruit` and keep `FruitRepository` read-only.

**Review by njr-11 — COMMENTED** [review]
Thanks for splitting the DELETE query tests into their own class.  I added a number of review comments.  One of them which applied across many tests is mixing `findAll()` into tests that do not otherwise perform querying without a `WHERE` clause.  This allows implementations that do not support queries without a `WHERE` clause to automatically pass your tests even if they have wrong behavior for what is intended to be tested, such as deletion based on equality of a single entity attribute with a value, and so forth.  I proposed updates to these to make the checking perform a query that is consistent with the DELETE query to reduce the chances of side-effects causing tests to incorrectly pass.

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java (hunk 25 above)
Comment (njr-11) [reply tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java:54]: Alternatively, the DELETE tests could continue using this FruitRepository which would no longer be read-only, and the SELECT tests could switch to one of our several other repositories that are read-only.

Comment (otaviojava) [reply tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java:54]: Guys, it would be amazing if we could read and write with **the same entity**, just as we usually do in the real world.

I would insist on keeping the same entity.

Comment (KyleAure) [reply tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaDeleteQueryTests.java:54]: You can, but for test performance / consistency it makes sense to split up the tests that do read-only operations from those that do read-write operations both into seperate test classes and using separate repositories. 
What if someone wanted to run the `JakartaDeleteQueryTests` and `JakartaQueryQueryTests` in parallel.  The way you have it written now would cause failures since a test in one could affect the results of the other.

**Comment by otaviojava** [comment]
> Thanks for splitting the DELETE query tests into their own class. I added a number of review comments. One of them which applied across many tests is mixing `findAll()` into tests that do not otherwise perform querying without a `WHERE` clause. This allows implementations that do not support queries without a `WHERE` clause to automatically pass your tests even if they have wrong behavior for what is intended to be tested, such as deletion based on equality of a single entity attribute with a value, and so forth. I proposed updates to these to make the checking perform a query that is consistent with the DELETE query to reduce the chances of side-effects causing tests to incorrectly pass.

@njr-11 good point, thanks!

I've updated.

Could you check it again?

**Review by njr-11 — APPROVED** [review]
Approving with the understanding that Kyle's comment about not using the same repository as read-only and update-capable can be addressed by switching the other test class, JakartaQueryTests, to use one of the several existing read-only repositories.  That seems like the most efficient solution.  That can be done under a separate PR.

**Review by KyleAure — APPROVED** [review]
